regnety/config/config.py

```diff
@@ -35,12 +38,109 @@ class RegNetYConfig:
     wm: float
 
 
+@dataclass
+class TrainConfig:
+    """
+    Dataclass of training configuration for RegNetY
+
+    Args:
+        optimizer: One of "sgd", "adam", "adamw"
+        base_lr: Base learning rate for training
+        warmup_epochs: Number of epochs used for warmup
+        warmup_factor: Gradual linear warmup factor
+        total_epochs: Number of training epochs
+        weight_decay: Weight decay to be used in optimizer
+        momentum: Momentum to be used in optimizer
+        lr_schedule: One of "constant" or "half_cos"
+        log_dir: Path to store logs
+        model_dir: Path to store model checkpoints
+        cache_dir:Path to cache parsed dataset
+    """
+
+    optimizer: str
+    base_lr: float
+    warmup_epochs: int
+    warmup_factor: float
+    total_epochs: int
+    weight_decay: float
+    momentum: float
+    lr_schedule: str
+    log_dir: str
+    model_dir: str
+    cache_dir: str
+
+
+@dataclass
+class PreprocessingConfig:
+    tfrecs_filepath: List[str]
+    batch_size: int
+    image_size: int
+    crop_size: int
+    resize_pre_crop: int
+    augment_fn: Union[str, Callable]
+    num_classes: int
+    cache_dir: str
+    color_jitter: bool
+    scale_to_unit: bool
+    scale_method: str
+
+
+def get_preprocessing_config(
+    tfrecs_filepath: List[str] = None,
+    batch_size: int = 1024,
+    image_size: int = 512,
+    crop_size: int = 224,
+    resize_pre_crop: int = 320,
+    augment_fn: Union[str, Callable] = "default",
+    num_classes: int = 1000,
+    cache_dir: str = "gs://ak-europe-train/cache/",
```

Comment:
Since we are disabling caching (and it makes sense because of the volume of the dataset) I think it's better to remove this one to clear any confusion.

Reply:
I kept it so it can be used for smaller datasets, specifically ImageNette. But makes sense, I'll remove it.